agent/tools/toolsdir.go

```diff
@@ -216,3 +219,37 @@ func ChangeAgentTools(dataDir string, agentName string, vers version.Binary) (*c
 	}
 	return tools, nil
 }
+
+// maybeReadTools will return tools based on the dataDir and current juju version.
+// If no agent is found based on the current host series, search for agent binaries
+// for different series and the same version/arch.
+//
+// TODO: (hml) 2018-02-22
+// Added for lp:bug1749201 - once agent binaries without the series in the name are
+// available, ReadTools() can move back to ChangeAgentTools()
+func maybeReadTools(dataDir string, currentVers version.Binary) (*coretools.Tools, error) {
+	tools, err := ReadTools(dataDir, currentVers)
+	if err == nil {
+		return tools, err
+	}
+	logger.Tracef("%s not found (%s)", SharedToolsDir(dataDir, currentVers), err)
+
+	hostOS := utilsSeries.MustOSFromSeries(utilsSeries.MustHostSeries())
+	potentialSeries := utilsSeries.OSSupportedSeries(hostOS)
+
+	potentialVers := currentVers
+	for _, series := range potentialSeries {
+		potentialVers.Series = series
+		_, err := ReadTools(dataDir, potentialVers)
+		if err == nil {
+			logger.Tracef("%s found", SharedToolsDir(dataDir, potentialVers))
+			err = fs.Copy(SharedToolsDir(dataDir, potentialVers), SharedToolsDir(dataDir, currentVers))
```

Comment:
you need to patch downloaded-tools.txt (or something) to update the series

Reply:
the version will be updated, but not the url